### PR TITLE
Fix crashes, upgrade to NetCore 3.1, update Nuget deps

### PR DESCRIPTION
**Pull request by yurikus**

### CODE REVIEW

File: src/git-istage/git-istage.csproj [hunk 1]
```diff
@@ -15,12 +15,14 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="LibGit2Sharp" Version="0.26.0-preview-0070" />
+    <PackageReference Include="LibGit2Sharp" Version="0.26.2" />
+
```
Comment (terrajobst) [inline src/git-istage/git-istage.csproj:19]: Why the blank line?

File: src/git-istage/Application.cs [hunk 2]
```diff
@@ -152,7 +152,11 @@ private void UpdateRepository()
             var changes = _viewStage
                 ? _repository.Diff.Compare<TreeChanges>(tipTree, DiffTargets.Index)
                 : _repository.Diff.Compare<TreeChanges>(null, true);
-            var paths = changes.Select(c => c.Path).ToArray();
+
+            var paths = changes
+                .Where(p => (p.Mode != Mode.SymbolicLink && p.OldMode != Mode.SymbolicLink))
+                .Select(c => c.Path).ToArray();
```
Comment (terrajobst) [inline src/git-istage/Application.cs:158]: Is skipping them entirely the right behavior? Are changes on files with junctions modelled as two entries? Or should we just map path?

File: src/git-istage/git-istage.csproj (hunk 1 above)
Comment (yurikus) [reply src/git-istage/git-istage.csproj:19]: For readability, separate one-line ref from one with children tags.

File: src/git-istage/Application.cs (hunk 2 above)
Comment (yurikus) [reply src/git-istage/Application.cs:158]: In my case, I junctioned 5GB directory with 100's K of files between git worktrees. Completely unnecessary comparison which was causing an exception from LibGit2Sharp ` .. change is a directory ..` or something along those lines

File: src/git-istage/git-istage.csproj (hunk 1 above)
Comment (terrajobst) [reply src/git-istage/git-istage.csproj:19]: Fair enough :-)

**Review by terrajobst — APPROVED** [review]

**Comment by yurikus** [comment]
Thanks!

**Comment by terrajobst** [comment]
I've deployed a new version, https://www.nuget.org/packages/git-istage/0.3.3

However, it will take a NuGet a bit to update the index, so `dotnet tool update` won't see it yet.

**Pull request merged**